rpcserver.go

```diff
@@ -3942,6 +3942,11 @@ func (r *rpcServer) ClosedChannels(ctx context.Context,
 func (r *rpcServer) LookupHtlc(ctx context.Context,
 	in *lnrpc.LookupHtlcRequest) (*lnrpc.LookupHtlcResponse, error) {
 
+	if !r.cfg.StoreFinalHtlcResolutions {
```

Comment:
What is the reason for adding this restriction? Maybe there is a use case for looking up historic resolutions even when the flag is disabled?

On the other hand, I can imagine this prevents confusion for users wondering why they never get a result without the flag being set.

Reply:
Mostly due to confusion. I think if you enable the flag and then disable it again then you cannot expect a reliable outcome, so might as disable it.

Meanwhile what do you think we change the name to `LookupFinalHtlc` to further avoid confusion?

Reply:
Yes, sounds like a good change. Or even `LookupHtlcResolution` or `LookupFinalHtlcResolution`?

Reply:
Cool added a new PR to do the job.